### PR TITLE
feat(bpdm-cert): created get endpoint for certificate query with bpn

### LINKED ISSUE #23: API: Request certificates for legal entity by BPNL-Id (or BPNS / BPNA)
As a user I want to request a certificate information based on the BPN L / S / A, so that I can verify that the business partner with certificates.

API-Call (get): https://cert-system.net/api/catena/certificate/{bpn}
Returns certificates for a legal entity including all sites assigned to a certificate

- [ ] Define the response body
- [ ] Make BPN L/S/A as mandatory

### PR DESCRIPTION
**Pull request by SujitMBRDI**
## Description
In this pull request, create new endpoint for fetching certificates belongs to provided business partner number.
The business partner can be of any type including either legal entity or site.

Fixes #23 #2 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by alexsilva-CGI — APPROVED** [review]

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/CertificateNotExists.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.exception
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.ResponseStatus
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
```
Comment (nicoprow) [inline src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/CertificateNotExists.kt:25]: If you use an exception handler for your exceptions then you don't need to define response statuses here as they are overwritten in the exception handler. Just FYI

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/InvalidBpnFormatException.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.exception
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.ResponseStatus
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
Comment (nicoprow) [inline src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/InvalidBpnFormatException.kt:25]: same here

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt [hunk 3]
```diff
@@ -56,4 +61,33 @@ class CertificateService(
...
+        return when {
+            bpn.startsWith("BPNL") -> {
+                val certificates = certificateRepository.findByBusinessPartnerNumber(bpn, pageRequest)
+                if (certificates.isEmpty) {
+                    throw CertificateNotExists("Legal Entity", bpn)
+                }
+                certificates.toPageDto(certificateMapping::toCertificateDocumentResponseDto)
+            }
+
+            bpn.startsWith("BPNS") -> {
```
Comment (nicoprow) [inline src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt:78]: I would expect such logic when inserting documents. Now there is a situation in which I put in a document with a wrong business partner number and I can't search for it.

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt [hunk 4]
```diff
@@ -56,4 +61,33 @@ class CertificateService(
...
+
+    fun getCertificatesByBpn(bpn: String, pageRequest: PageRequest): PageDto<CertificateDocumentResponseDto> {
+        if (bpn.isBlank())
+            throw IllegalArgumentException("Provided business partner number is null or empty")
+
+        return when {
+            bpn.startsWith("BPNL") -> {
+                val certificates = certificateRepository.findByBusinessPartnerNumber(bpn, pageRequest)
+                if (certificates.isEmpty) {
+                    throw CertificateNotExists("Legal Entity", bpn)
```
Comment (nicoprow) [inline src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt:73]: These checks for empty page leads to a bit of a strange behaviour in the API. When I paginate to a page above the limit I then get this certificate does not exist error. However, the certificates   for this BPN actually exist in the previous pages.
I would rather expect to just get an empty page or at least get a matching error message that I paginated over the total element collection.

**Review by nicoprow — CHANGES_REQUESTED** [review]
Some strange behaviours on the API

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/CertificateNotExists.kt (hunk 1 above)
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/exception/CertificateNotExists.kt]: Yeah, removed ResponseStatus

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt (hunk 3 above)
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt:79]: Yeah, i am planning to add all validation including business partner number, site details also registration number into the new pull request. All validation will be conducted while creating Certificate entity together.

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt (hunk 4 above)
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt:74]: Thanks for detailed explanation, i have adapted changes in same pull request.

**Review by nicoprow — APPROVED** [review]

**Pull request merged**